OverloaderAIDriver.lua

```diff
@@ -27,17 +27,20 @@ OverloaderAIDriver.myStates = {
 }
 
 function OverloaderAIDriver:init(vehicle)
+    --there seems to be a bug, where "vehicle" is not always set once start is pressed
+	if vehicle and vehicle:isa(Vehicle) then
+		self:findPipeAndTrailer(vehicle)
+	end
 	CombineUnloadAIDriver.init(self, vehicle)
     self:initStates(OverloaderAIDriver.myStates)
     self:debug('OverloaderAIDriver:init()')
     self.mode = courseplay.MODE_OVERLOADER
-    self.unloadCourseState = self.states.ENROUTE
-    self:findPipeAndTrailer()
+	self.unloadCourseState = self.states.ENROUTE
     self.nearOverloadPoint = false
 end
 
-function OverloaderAIDriver:findPipeAndTrailer()
-    local implementWithPipe = AIDriverUtil.getImplementWithSpecialization(self.vehicle, Pipe)
+function OverloaderAIDriver:findPipeAndTrailer(vehicle)
```

Comment:
but isn't this only because we call it before CombineUnloadAIDriver.init(self, vehicle), which initializes self.vehicle?

Reply:
`findPipeAndTrailer()` has to be called before  `OverloaderAIDriver:setHudContent()` so `courseplay.hud:setOverloaderAIDriverContent(self.vehicle)` can be called